src/Error/Error.php

```diff
@@ -47,28 +47,43 @@
     private $filePath;
 
     /**
-     * @param int    $type
-     * @param string $filePath
+     * @var \Exception|\Error|\Throwable|null
      */
-    public function __construct($type, $filePath)
+    private $source;
+
+    /**
+     * @param int                               $type
+     * @param string                            $filePath
+     * @param \Exception|\Error|\Throwable|null $source
```

Comment:
I agree we don't write code for IDEs too, but as `\Throwable|null` is valid and is well supported, there is no need to use an alternative syntax IMO.